### ISSUE TITLE
Testing connection to extension server failed & the database needs to be updated/repaired

### ISSUE THREAD

**Issue by SchneiderDe** [bolt/bolt#7185]
Hello,

Using 3.4.1 installed through composer. It is a fresh installation on windows 7. I get two errors:

Testing connection to extension server failed.
The database needs to be updated/repaired.

A connection should me made as there is no outbound restriction of connection. Database does not get repaired at all. Such an error was displayed in earlier versions. Now it appeared again. Is there something I could do to remove this or it is a bug, like it was before?

**Comment by bobdenotter** [comment]
Hi @SchneiderDe,

Without more details, there's not much concrete info we can give you. I'm guessing you're not using a "local" server, but a remote one, right? 

Please fill out the relevant details in the [Issue template](https://github.com/bolt/bolt/blob/3.4/.github/ISSUE_TEMPLATE.md), which we wrote for this very purpose.

**Comment by SchneiderDe** [comment]
Hello,

Thanks for the reply. 

A simple answer to your assumption: I am not using localhost to connect to local URI/bolt under XAMP. I am using a local server, which is NOT localhost but a Domain.lan that gets resolved through an internal DNS server within the network. The network DNS server has IP mapping of domain.lan, which functions - technically - no different than all internet DNSservers.  

The fact should be noticed, that I can reach admin pages under Domain.lan shows that there could be no problem with DNS resolver. The issue is definitely not here. If it would not resolve, then respective php scripts would through an error to show that the page is not reachable.

The file attributes had archive stamp on them, which normally happens on windows 7. Nothing unusual. I did remove them too.

I have used the latest XAMP for php 7.1. Here are the issue template details:
```
Relevant Bolt Version: [ 3.4.1  | master ]
Install type: [ Composer install ]
PHP version: [ 7.1]
Used web server: [ Apache ] version [ 2.4 ]
For UX/UI issues: [ Waterfox ]
OS: Windows 7 64bits.
```
Assuming that it cannot be related to the server environment, I suspect that this problem may be arising from javascripts or php scripts. For e.g. in *.js (?), there are entries of market.bolt.cm. Somehow, the scripts fails to make an outbound connection. Further, I used SQLite. Even here, it should not have any problem to connect the db and make changes.

As I use php 7.1, it is possible that something did not function under windows as expected, most likely some javascript of php script. Thus, a likely area would be to see the scripts that makes URI related to db and a TCP connection to extension to remote server. If you want, I have no problem to test it with MySQl, which is further available for testing. Although I think the problem could not be a connection to db.

**Comment by GawainLynch** [comment]
The connection problem is your local network set up. The connection is done in PHP via Guzzle, and that error occurs when there is a DNS problem or timeout. At a guess, the PHP installation on the Windows host is not using the correct DNS configuration.

The database update problem with Sqlite is due to a bug in Doctrine DBAL, see this PR that is pending upstream https://github.com/doctrine/dbal/pull/2865

**Comment by SchneiderDe** [comment]
Hi GawainLynch,

Vow, thats a very nice guess. Thanks for that hint regarding the db update problem. I will have to wait for that.

I understand that the connection is done in PHP via Guzzle. Here, a common thing is the Windows host configuration amongst all other php scripts. There is not a single php script, or any other software, which has problems with the Windows host config. Had it been the case, it would show up somewhere, in composer, in whois queries, in one of many other scripts. All the Windows host configuration is done that is possible at all, like resolve.conf, DNS resolver, etc....

Until I installed bolt, I did not notice any of the Windows host problem with other software or php scripts. Here, I have configured everywhere it needed to be properly, I assume. So now I do believe that the combination of PHP via Guzzle together with the Windows host is perhaps the problem area.

Is there any other hint from anyone where I should look for a potential problem or even investigate further to provide more infos on it? I saw the following but it did not really say much:

https://stackoverflow.com/questions/20847633/limit-connecting-time-with-guzzle-http-php-client
https://github.com/guzzle/guzzle/issues/590
http://docs.guzzlephp.org/en/stable/faq.html#does-guzzle-require-curl

May be the scripts in bolt that invoke a connection to PHP via Guzzle could be revisited to see if something is in there that could be changed perhaps... Otherwise, I will have to wait until someone else comes up with a solution or report similar problem...
